app/models/miq_queue.rb

```diff
@@ -237,7 +237,12 @@ def self.put_or_update(find_options)
     #   it's unordered nature.
     where_scope = if conds.key?(:args)
                     args = YAML.dump conds.delete(:args)
-                    MiqQueue.where(conds).where(['args = ?', args])
```

Comment:
also feels like we could just use `conds` and not have to add this junk.

maybe just put the `YAML.dump` version back into there.

by the way, I think this conversation is out of scope for @enoodle and belongs in a separate PR